selfdrive/ui/qt/sidebar.cc

```diff
@@ -100,9 +100,17 @@ void Sidebar::updateState(const UIState &s) {
   setProperty("tempStatus", QVariant::fromValue(tempStatus));
 
   ItemStatus pandaStatus = {{tr("VEHICLE"), tr("ONLINE")}, good_color};
+  bool gpsOK = true;
+  for (auto event : sm["onroadEvents"].getOnroadEvents()) {
+    if (event.getName() == cereal::CarEvent::EventName::NO_GPS) {
+      gpsOK = false;
+      break;
+    }
+  }
+
   if (s.scene.pandaType == cereal::PandaState::PandaType::UNKNOWN) {
     pandaStatus = {{tr("NO"), tr("PANDA")}, danger_color};
-  } else if (s.scene.started && !sm["liveLocationKalman"].getLiveLocationKalman().getGpsOK()) {
+  } else if (s.scene.started && !gpsOK) {
```

Comment:
Should use what controlsd uses `self.gps_location_service` so that the sidebar accurately reflects if there's a GPS fix. With this it will show a GPS fix immediately, then show SEARCH 1.5km later (if it never got a fix)

Reply:
removed it from the sidebar, since it wont be useful to know about it because nothing will be using it after locationd refactor is merged (https://github.com/commaai/openpilot/pull/33029)